src/Umbraco.Web.UI.Client/src/views/propertyeditors/datepicker/datepicker.controller.js

```diff
@@ -78,9 +78,20 @@ function dateTimePickerController($scope, notificationsService, assetsService, a
         setDatePickerVal();
     };
 
-    $scope.inputChanged = function() {
-        setDate($scope.model.datetimePickerValue);
-        setDatePickerVal();
+    $scope.inputChanged = function () {        
+        if ($scope.model.datetimePickerValue == "" && $scope.hasDatetimePickerValue) {
```

Comment:
Can you explain this one? If the value on the model is empty AND `hasDatetimePickerValue` is true? When would this be the case?

Also makes sure to use a triple equals there: `===`

Reply:
Yeah, in hindsight it could do with at least a comment to explain what it's checking for. `$scope.hasDatetimePickerValue` hasn't been updated yet, so it tells us whether we had a value before the text box changed. So the test makes sure that we only call `$scope.clearDate` if there's a previous date to clear. This matters because clearDate will mark the form as dirty. If you start with a blank date and then trigger inputChanged with an empty value, we don't want that to trigger "unsaved changes" messages.